### PR TITLE
unittests in test_fpp.py

### PR DESCRIPTION
**Pull request by adenjonah**
I created tests for all non-graphical functions in fpp and verified intended output.

I used assertAlmostEqual due to Python decimal arithmetic.

I also added an if name == main at the end of this test file to allow for easier running of the tests from a terminal.

Lastly, I had to create a virtual environment to get statwrap working locally on my machine, so I added all newly created venv and cache files to the gitignore.

### CODE REVIEW

File: tests/test_fpp.py [hunk 1]
```diff
@@ -1,20 +1,242 @@
...
+        self.numbers_multiple = [1, 2, 3, 4, 5]
+        self.single_value = [42]
+        self.mixed_types = [1, 2.5, 3, 4.5]
+
+    def test_average_array(self):
+        result = average(self.numbers_array)
+        expected = 0.2
+        self.assertAlmostEqual(result, expected)
+
+    def test_average_multiple_args(self):
```
Comment (alexanderthclark) [inline tests/test_fpp.py:48]: can you include a comment to make the difference between this and the test on 43 more apparent, where this one unpacks a list and the other doesn't?

File: tests/test_fpp.py [hunk 2]
```diff
@@ -1,20 +1,242 @@
...
+        pass
+
+class TestSDPlus(unittest.TestCase):
+
+    def setUp(self):
+        self.numbers_array = [-1, 0, 1]
+        self.numbers_multiple = [1, 2, 3, 4, 5]
+        self.single_value = [42]
+        self.mixed_types = [1, 2.5, 3, 4.5]
+        self.zeroes = [0, 0, 0, 0]
```
Comment (alexanderthclark) [inline tests/test_fpp.py:163]: let's add a test for `self.singleton = [1]` where that should give a zerodivision or another error.

**Review by alexanderthclark — CHANGES_REQUESTED** [review]
Looks good! Just a couple comments. 

edit: Can you also just confirm in the PR that you were able to successfully install the package locally?

**Comment by adenjonah** [comment]
image
image

- Added comment
- Made vars_plus and sd_plus raise an error when n <= 1
- Rewrote documentation to reflect that change (pictured above)
- Added tests for testing the error functionality
- These changes are in the following commit

**Comment by adenjonah** [comment]
Lastly, to confirm I was able to install the package locally

**Review by alexanderthclark — APPROVED** [review]
lgtm

**Pull request merged**